### ISSUE TITLE
Regexp-based highlighting of lines

### ISSUE THREAD

**Issue by soegaard** [greghendershott/racket-mode#700]
Hi,

I am using racket-mode to edit text written in a Scribble-inspired format.

It looks something like this:
```
@chapter{Vektorer i planen}
@table-of-contents[2]{}

@section{Vektor}
@index{vektor}@index{forskydning}@index{hastighed}@index{fart}@index{knob}@index{pil}
@index{vektor, definition}@index{repræsentant}@index{skalar}@index{nulvektor}

I dette kapitel skal vi se på vektorer. Vektorer bruges til at beskrive
objekter, der har en længde og en retning. Begrebsmæssigt er en 
vektor er karakteriseret ved en @it{længde} og en @it{retning}. 
```

When the document is large, it helps to have chapters, sections and subsections stand out.

I have discovered that `highlight-regexp` can be used to highlight lines that matches a regexp.
It looks something like:

<img width="802" alt="image" src="https://github.com/greghendershott/racket-mode/assets/461765/8f9dafcd-1a92-45b3-bddd-5c27758bd4c0">

But ... this removes the highlighting from `racket-mode`. 

So the question whether it is possible to have an `racket-mode-highlight-regexp` that sets the background color
of a line, if it matches a regexp - and otherwise keep the normal highlighting computed by `racket-mode`.

**Comment by greghendershott** [comment]
I think it matters what face you tell it to use -- and what that face specifies.

When I try this and choose a face like `hl-line`, it works as you want (IIUC).

This is because the `hl-line` face specifies a `background` property -- but _not_ any `foreground` property (the value is "undefined' if you `M-x describe-face`).

Whereas if I try with most faces, which _do_ specify a foreground property, it works in the undesired way that you report.

TLDR, AFAICT: You want a face that supplies only the thing you want -- a particularly background -- but nothing else.

Does that help?

**Comment by soegaard** [comment]
Yes, that solves it. Much simpler than I feared.
I just need to stick to a face, that only specifies a background property.

For others that find this thread:

    M-x list-faces-display
 
Thanks!
